### ISSUE TITLE
MSSQL spike - connect MSSQL database to HL

### ISSUE THREAD

**Issue by RobStallion** [healthlocker/healthlocker#837]
- [x] Set up MSSQL database

- [ ] Set up MSSQL adapter 

- [ ] Configure for MSSQL 

See #782 for more details

**Comment by katbow** [comment]
To set up MSSQL database on mac, we need to [run the SQL Server on a Linux container on Docker](https://www.microsoft.com/en-gb/sql-server/sql-server-2017#resources).

**Comment by katbow** [comment]
The plan instead for this is to sign up to free microsoft azure trial, then use [node.js to connect to the azure sql db](https://docs.microsoft.com/en-us/azure/sql-database/sql-database-connect-query-nodejs). 

After we have this working, then we can work on connecting with Elixir.

**Comment by katbow** [comment]
Update: We've connected using node and were able to select, insert, and update the mssql db. 

Next step is to attempt connecting with elixir.

**Comment by RobStallion** [comment]
Update

Have been trying to replicate connecting to the database with elixir but have been running into a number of issues. 

We configured the database following the [instructions provided by mssql_ecto](https://github.com/findmypast-oss/mssql_ecto). MssqlEcto is an Ecto Adapter for [Mssqlex](https://github.com/findmypast-oss/mssqlex), which is an adapter to Microsoft SQL Server (using DBConnection and ODBC). 

We get the following error in the terminal after we set this up..

![image](https://user-images.githubusercontent.com/15571853/27092865-b83795a8-505c-11e7-821e-5a8218701ef6.png)

and this error when we go to localhost..

![image](https://user-images.githubusercontent.com/15571853/27093130-78dc3bf6-505d-11e7-9678-b807da320925.png)

Have spend a few hours looking into causes for the problem and it seems to be related to odbc. It appears the newer versions of Mac OSX (versions from 10.9 onward) may not include unixodbc, which Erlang ODBC relies on.

Have found a number of articles, [most helpful being this one](http://stratus3d.com/blog/2014/10/24/install-erlang-16-on-mac-osx-10-dot-9-with-kerl/), where people have had similar issues and have tried to follow instructions provided but so far have had no luck.

**Comment by RobStallion** [comment]
Update

Have managed to install odbc but we are now getting a new error..
```
[error] Mssqlex.Protocol (#PID<0.9753.1>) failed to connect: ** (Mssqlex.Error) odbc_not_started
```

Currently looking into solutions and have opened [an issue]() on the mssql_ecto repo to see if there is any advice that they could provide.

**Comment by RobStallion** [comment]
Following the instructions in this guide:
https://docs.microsoft.com/en-gb/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server

**Comment by nelsonic** [comment]
@RobStallion please describe where you got to and what you plan to try next (_as discussed_) thanks!

**Comment by RobStallion** [comment]
Update

Have created a repo here: https://github.com/RobStallion/azure-elixir which contains everything that we have tried up until this point.

Have also commented on an issue here: https://github.com/findmypast-oss/mssql_ecto/issues/15, and linked my new repo in that comment, explaining the situation and where we currently are with it.

**Comment by RobStallion** [comment]
Update:

Trying to connect [this repo]( https://github.com/RobStallion/azure-elixir) with travis to check if we are having a problem connecting with any ms sql db or just the one on azure.

**Comment by RobStallion** [comment]
It appears that travis does not support running a virtual box in their environment as stated in this issue: https://github.com/travis-ci/travis-ci/issues/6060

**Comment by katbow** [comment]
Here's a full run through of what we've done for this. It's outlined above but I wanted to clarify anything we've missed and provide more detail:

1. Started setting up docker to run a SQL server in a Linux container. We were advised to instead set up an MSSQL db on Microsoft Azure (test database, not HL ePJS at this point simply to test connectivity).
2. Signed up to Microsoft Azure & set up a test database to be used.
3. Connected to Microsoft Azure test database using node. Proves that we can connect & the database server is live.
4. When connecting to the database using elixir there were a number of problems we encountered. 
    * [First the odbc application is not included by default in macOS erlang installs](https://github.com/healthlocker/healthlocker/issues/837#issuecomment-308179663)
    * We managed to install erlang with ODBC application by running `brew install freetds`, `brew install unixodbc`, then uninstalling erlang and installing again with `brew install erlang -v --with-native-libs -y`
    * After installing Erlang with odbc, we had an error when we ran the server that odbc was not started
    * We discovered & commented on [this closed issue](https://github.com/findmypast-oss/mssql_ecto/issues/14), which was the same error but due to an incorrect version of Elixir (we had the correct version, however). 
    * The maintainers requested we open a [new issue](https://github.com/findmypast-oss/mssql_ecto/issues/15) which has some suggestions that we tried locally, but did not work.
    * We set up [Vagrant](https://github.com/dwyl/learn-vagrant), and ran our node script to ensure it worked in the ubuntu environment
    * After this worked, we set up our vagrant file for running an elixir project on ubuntu
    * We were able to run an elixir project (without using MSSQL, just a static page in elixir)
    * After updating the vagrant file to include the drivers & erlang versions needed for ODBC, we encountered the same error when running locally on macOS (`odbc_not_started`)
    * We were going to try running the code on travis, however it's not supported as mentioned by @RobStallion in the comment above. 
    * Next steps are to create a node script which will query the database, then get that information back in elixir.

**Comment by RobStallion** [comment]
closed in favour of #843
